### ISSUE TITLE
[Bug] ryujinx not working on radv on 1.1.805+

### ISSUE THREAD

**Issue by A-w-x** [Ryujinx/Ryujinx#5143]
### Description of the issue

glitches and crashes in multiple games

a few:
legend arceus
scarlet\violet
links awakening

### Reproduction steps

run games

### Log file

an example shot from scarlet
![Screenshot_20230529_135709](https://github.com/Ryujinx/Ryujinx/assets/80461138/04895f46-5887-43d2-b830-c7238a7ad6d5)

log from links awakening
[Ryujinx_1.1.805_2023-05-29_13-57-39.log](https://github.com/Ryujinx/Ryujinx/files/11591543/Ryujinx_1.1.805_2023-05-29_13-57-39.log)

### OS

archlinux

### Ryujinx version

1.1.805 - current

### Game version

any

### CPU

ryzen 6800U

### GPU

radeon 680m

### RAM

32gb

### List of applied mods

none

**Comment by 0x647262** [comment]
> glitches and crashes in multiple games

For crashes specifically, is there any information in your dmesg log (`journalctl --dmesg --grep amdgpu`)? I'm having similar issues and when a game outright crashes I get:

```
May 29 22:18:01 onyx kernel: amdgpu 0000:0f:00.0: amdgpu: [gfxhub] page fault (src_id:0 ring:24 vmid:2 pasid:32769, for process Ryujinx pid>
May 29 22:18:01 onyx kernel: amdgpu 0000:0f:00.0: amdgpu:   in page starting at address 0x000080010b8f8000 from client 0x1b (UTCL2)
May 29 22:18:01 onyx kernel: amdgpu 0000:0f:00.0: amdgpu: GCVM_L2_PROTECTION_FAULT_STATUS:0x00201031
May 29 22:18:01 onyx kernel: amdgpu 0000:0f:00.0: amdgpu:          Faulty UTCL2 client ID: TCP (0x8)
May 29 22:18:01 onyx kernel: amdgpu 0000:0f:00.0: amdgpu:          MORE_FAULTS: 0x1
May 29 22:18:01 onyx kernel: amdgpu 0000:0f:00.0: amdgpu:          WALKER_ERROR: 0x0
May 29 22:18:01 onyx kernel: amdgpu 0000:0f:00.0: amdgpu:          PERMISSION_FAULTS: 0x3
May 29 22:18:01 onyx kernel: amdgpu 0000:0f:00.0: amdgpu:          MAPPING_ERROR: 0x0
May 29 22:18:01 onyx kernel: amdgpu 0000:0f:00.0: amdgpu:          RW: 0x0
...
May 29 22:18:11 onyx kernel: amdgpu 0000:0f:00.0: amdgpu: IH ring buffer overflow (0x000A3EA0, 0x00023E60, 0x00023EC0)
May 29 22:18:11 onyx kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx_0.0.0 timeout, signaled seq=19278, emitted seq=19280
May 29 22:18:11 onyx kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* Process information: process Ryujinx pid 2873 thread GUI.RenderLoop>
May 29 22:18:11 onyx kernel: amdgpu 0000:0f:00.0: amdgpu: GPU reset begin!
May 29 22:18:11 onyx kernel: amdgpu 0000:0f:00.0: amdgpu: Guilty job already signaled, skipping HW reset
May 29 22:18:11 onyx kernel: amdgpu 0000:0f:00.0: amdgpu: GPU reset(2) succeeded!
...
```

As described here https://gitlab.freedesktop.org/drm/amd/-/issues/2496

**Comment by A-w-x** [comment]
just checked
yes, it all started in build 805
![Screenshot_20230603_102227](https://github.com/Ryujinx/Ryujinx/assets/80461138/262e2510-1ab2-4fb3-8f4c-c2cb4c020186)

**Comment by 0x647262** [comment]
Thanks!

Build 804 works for me as well. I'll bisect this and report the specific commit in the Mesa issue since this is likely a RADV bug instead of a Ryujinx bug.

**Comment by piplup55** [comment]
> Thanks!
> 
> Build 804 works for me as well. I'll bisect this and report the specific commit in the Mesa issue since this is likely a RADV bug instead of a Ryujinx bug.

i don't think it is, i tried 22.3.6 from the arch repos and it still crashed

**Comment by piplup55** [comment]
## Games afflicted
all games i've noticed anyway
- Metroid Dread (System Lockup/ErrorDeviceLost)
- Mario 3D All Stars (System Lockup/ErrorDeviceLost)
- Mario + Rabbids Kingdom Battle (System Lockup/ErrorDeviceLost)
- Kirby and the forgotten land (System Lockup/ErrorDeviceLost)
- Pokemon Sword/Shield (ErrorDeviceLost)
- Pokemon Scarlet/Violet (ErrorDeviceLost during compiling shaders)
- Calculator (ErrorDeviceLost)

**Comment by TSRBerry** [comment]
Seems to be a regression from: f864a490142c4da608dab8d2025fc18da857f93f

**Comment by gdkchan** [comment]
Fixed by #5267.
